stdlib/number.gr

```diff
@@ -493,3 +939,107 @@ export let cos = (radians: Number) => {
     sin(pi / 2 + radians)
   }
 }
+
+/**
+ * Computes the tangent of a number (in radians) using Chebyshev polynomials.
+ *
+ * @param radians: The input in radians
+ * @returns The computed tangent
+ *
+ * @since v0.5.4
+ */
+export let tan = (radians: Number) => {
+  if (isNaN(radians) || isInfinite(radians)) {
+    nan
+  } else {
+    sin(radians) / cos(radians)
+  }
+}
+
+// Math.gamma implemented using the stirling's approximation
+// https://en.wikipedia.org/wiki/Lanczos_approximation
+/**
+ * Computes the gamma function of a value using
+ *
+ * @param z: The value to compute the gamma of
+ * @returns The gamma of the given value
+ *
+ * @since v0.5.4
+ */
+export let rec gamma = z => {
+  if (z == 0) {
+    fail "Gamma of 0 is undefined"
```

Comment:
As a note we use fail throughout the Number library over exceptions so if we use a real exception there it would make sense to use it elsewhere in the number library wouldn't it?

Reply:
Real exceptions are better than strings for sure!